quark.c

```diff
@@ -392,6 +392,17 @@ event_storage_clear(struct quark_queue *qq)
 		free(qq->event_storage.shm);
 		qq->event_storage.shm = NULL;
 	}
+
+	if (qq->event_storage.tty != NULL) {
+		struct quark_tty *current = qq->event_storage.tty;
+		struct quark_tty *next = current->next;
+		while (next != NULL)
```

Comment:
```suggestion
		struct quark_tty *current;
		struct quark_tty *next = qq->event_storage.tty->next;
		while (next != NULL) {
```

This is all correct, I'd just suggest changing the assignment a bit, to make it clear you never actually free (qq->event_storage.tty) here, also the assignment is just reassigned below.

Reply:
Makes sense.

3c02cc86f262e56a73857a1cc8e49a02a45a0fe3